src/webots/control/WbLanguageTools.cpp

```diff
@@ -102,34 +104,42 @@ QString WbLanguageTools::pythonCommand(QString &shortVersion, const QString &com
   } else
     shortVersion = QString(version[0][0]) + version[0][2];
 #elif __APPLE__
-  if (pythonCommand == "python" || pythonCommand == "python3") {
-    pythonCommand = findWorkingPythonPath("3.8", env, false);
-    shortVersion = "38";
-    if (pythonCommand == "!") {
-      pythonCommand = findWorkingPythonPath("3.9", env, false);
-      shortVersion = "39";
+  if (std::getenv("PWD")) {
+    shortVersion = checkIfPythonCommandExist(pythonCommand, env, true);
+    if (shortVersion.isEmpty()) {
+      pythonCommand = "!";
+      WbLog::warning(QObject::tr("Python was not found.\n") + advice);
+    }
+  } else {
+    if (pythonCommand == "python" || pythonCommand == "python3") {
+      pythonCommand = findWorkingPythonPath("3.8", env, false);
+      shortVersion = "38";
       if (pythonCommand == "!") {
-        pythonCommand = findWorkingPythonPath("3.7", env, true);
-        shortVersion = "37";
+        pythonCommand = findWorkingPythonPath("3.9", env, false);
+        shortVersion = "39";
+        if (pythonCommand == "!") {
+          pythonCommand = findWorkingPythonPath("3.7", env, true);
+          shortVersion = "37";
+        }
       }
+    } else if (pythonCommand == "python3.7") {
+      pythonCommand = findWorkingPythonPath("3.7", env, true);
+      shortVersion = "37";
+    } else if (pythonCommand == "python3.8") {
+      pythonCommand = findWorkingPythonPath("3.8", env, true);
+      shortVersion = "38";
+    } else if (pythonCommand == "python3.9") {
+      pythonCommand = findWorkingPythonPath("3.9", env, true);
+      shortVersion = "39";
+    } else {
+      shortVersion = checkIfPythonCommandExist(pythonCommand, env, true);
+      if (shortVersion.isEmpty())
+        pythonCommand = "!";
     }
-  } else if (pythonCommand == "python3.7") {
-    pythonCommand = findWorkingPythonPath("3.7", env, true);
-    shortVersion = "37";
-  } else if (pythonCommand == "python3.8") {
-    pythonCommand = findWorkingPythonPath("3.8", env, true);
-    shortVersion = "38";
-  } else if (pythonCommand == "python3.9") {
-    pythonCommand = findWorkingPythonPath("3.9", env, true);
-    shortVersion = "39";
-  } else {
-    shortVersion = checkIfPythonCommandExist(pythonCommand, env, true);
-    if (shortVersion.isEmpty())
-      pythonCommand = "!";
-  }
 
-  if (pythonCommand == "!")
-    WbLog::warning(QObject::tr("Python was not found.\n") + advice);
+    if (pythonCommand == "!")
+      WbLog::warning(QObject::tr("Python was not found.\n") + advice);
```

Comment:
You could keep this outside the if/else block and write it only once in stead of duplicating it for the "PWD" case.